js/main.js

```diff
@@ -0,0 +1,294 @@
+
+  var $castContainer = document.createElement('div');
+  $castContainer.className = 'row align-center';
+  $filmInfo.appendChild($castContainer);
+
+  var $cast = document.createElement('p');
+  $cast.className = 'cast-writer-style-white';
+  $cast.textContent = 'Cast';
+  $castContainer.appendChild($cast);
+
+  var $actorsList = document.createElement('p');
+  $actorsList.className = 'cast-writer-style-blue';
+  $actorsList.textContent = movie.cast;
+  $castContainer.appendChild($actorsList);
+
+  var $writersContainer = document.createElement('div');
+  $writersContainer.className = 'row align-center';
+  $filmInfo.appendChild($writersContainer);
+
+  var $writers = document.createElement('p');
+  $writers.className = 'cast-writer-style-white';
+  $writers.textContent = 'Writers';
+  $writersContainer.appendChild($writers);
+
+  var $writersList = document.createElement('p');
+  $writersList.className = 'cast-writer-style-blue';
+  $writersList.textContent = movie.writer;
+  $writersContainer.appendChild($writersList);
+}
+
+function searchTitle(event) {
+  event.preventDefault();
+  var xhr = new XMLHttpRequest();
+  var searchTitle = $titleForm.title.value;
+  var searchYear = $titleForm.year.value;
+  var movieEntry = {};
+
+  if (searchYear !== '') {
+    xhr.open('GET', 'http://www.omdbapi.com/?t=' + searchTitle + '&' + 'y=' + searchYear + '&' + 'plot=full' + '&' + apikey);
+  } else {
+    xhr.open('GET', 'http://www.omdbapi.com/?t=' + searchTitle + '&' + 'plot=full' + '&' + apikey);
+  }
+  xhr.responseType = 'json';
+
+  function createSearchResult(event) {
+    movieEntry.title = xhr.response.Title;
+    movieEntry.year = xhr.response.Year;
+    movieEntry.director = xhr.response.Director;
+    movieEntry.writer = xhr.response.Writer;
+    movieEntry.cast = xhr.response.Actors;
+    movieEntry.plot = xhr.response.Plot;
+    movieEntry.imdbID = xhr.response.imdbID;
+    movieEntry.poster = xhr.response.Poster;
+    movieEntry.runtime = xhr.response.Runtime;
+
+    renderSearchResult(movieEntry);
+    changeView('search-result');
+  }
+  xhr.addEventListener('load', createSearchResult);
+  xhr.send();
+}
+
+$titleForm.addEventListener('submit', searchTitle);
+
+function searchID(event) {
+  event.preventDefault();
+  var xhr = new XMLHttpRequest();
+  var searchID = $idForm.imdbID.value;
+  var movieEntry = {};
+
+  xhr.open('GET', 'http://www.omdbapi.com/?i=' + searchID + '&' + apikey);
+  xhr.responseType = 'json';
+
+  function createSearchResult(event) {
+    movieEntry.title = xhr.response.Title;
+    movieEntry.year = xhr.response.Year;
+    movieEntry.director = xhr.response.Director;
+    movieEntry.writer = xhr.response.Writer;
+    movieEntry.cast = xhr.response.Actors;
+    movieEntry.plot = xhr.response.Plot;
+    movieEntry.imdbID = xhr.response.imdbID;
+    movieEntry.poster = xhr.response.Poster;
+    movieEntry.runtime = xhr.response.Runtime;
+
+    renderSearchResult(movieEntry);
+    changeView('search-result');
+  }
```

Comment:
This function is exactly the same as the other function by this name in the above function. In general, I suggest you only define functions in the global scope. That will make it so that you can reuse this function as well. If you are doing this to have access to `xhr`, you should just give this function a parameter, and pass the response as an argument to the function.